ldi-orchestrator/ldio-application/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/config/FlowAutoConfiguration.java

```diff
@@ -3,70 +3,82 @@
+		this.configContext = configContext;
 	}
 
-	@Bean
-	@ConditionalOnMissingBean(LdiInput.class)
-	@DependsOn({ "componentExecutor" })
-	public LdiInput ldtoInput(OrchestratorConfig orchestratorConfig) {
-		return (LdiInput) getLdiComponent(orchestratorConfig.getInput().getName(),
-				orchestratorConfig.getInput().getConfig());
+	@PostConstruct
+	public void registerInputBeans() {
+		config.getPipelines().forEach(this::initialiseLdiInput);
 	}
 
-	@Bean("componentExecutor")
-	public ComponentExecutor componentExecutor(final OrchestratorConfig orchestratorConfig) {
-		List<LdiTransformer> ldiTransformers = orchestratorConfig.getTransformers()
+	public ComponentExecutor componentExecutor(final PipelineConfig pipelineConfig) {
+		List<LdiTransformer> ldiTransformers = pipelineConfig.getTransformers()
 				.stream()
-				.map(this::ldtoTransformer)
+				.map(this::getLdioTransformer)
 				.toList();
-		List<LdiOutput> ldiOutputs = orchestratorConfig.getOutputs()
+		List<LdiOutput> ldiOutputs = pipelineConfig.getOutputs()
 				.stream()
-				.map(this::ldtoOutput)
+				.map(this::getLdioOutput)
 				.toList();
 		return new ComponentExecutorImpl(ldiTransformers, ldiOutputs);
 	}
 
-	@Bean
-	public LdiAdapter ldiAdapter(final OrchestratorConfig orchestratorConfig) {
-		ComponentDefinition adapterDefinition = orchestratorConfig.getInput().getAdapter();
-
-		return (LdiAdapter) getLdiComponent(adapterDefinition.getName(), adapterDefinition.getConfig());
-	}
-
-	private LdiTransformer ldtoTransformer(ComponentDefinition componentDefinition) {
+	private LdiTransformer getLdioTransformer(ComponentDefinition componentDefinition) {
 		return (LdiTransformer) getLdiComponent(componentDefinition.getName(), componentDefinition.getConfig());
 	}
 
-	private LdiOutput ldtoOutput(ComponentDefinition componentDefinition) {
+	private LdiOutput getLdioOutput(ComponentDefinition componentDefinition) {
 		return (LdiOutput) getLdiComponent(componentDefinition.getName(), componentDefinition.getConfig());
 	}
 
+	public void initialiseLdiInput(PipelineConfig config) {
+		SingletonBeanRegistry beanRegistry = configContext.getBeanFactory();
+		LdioInputConfigurator configurator = (LdioInputConfigurator) configContext.getBean(
+				config.getInput().getName());
+		LdiAdapter adapter = (LdiAdapter) getLdiComponent(config.getInput().getAdapter().getName(),
+				config.getInput().getAdapter().getConfig());
+		ComponentExecutor executor = componentExecutor(config);
+
+		String pipeLineName = config.getName();
+
+		Map<String, String> inputConfig = new HashMap<>(config.getInput().getConfig().getConfig());
+		inputConfig.put("pipeline.name", pipeLineName);
+
+		Object ldiInput = configurator.configure(adapter, executor, new ComponentProperties(inputConfig));
+
+		if (!beanRegistry.containsSingleton(pipeLineName)) {
+			beanRegistry.registerSingleton(pipeLineName, ldiInput);
+		}
+	}
+
 	private LdiComponent getLdiComponent(String beanName, ComponentProperties config) {
```

Comment:
not really related to this pr, but I wonder what additional benefit the LdiComponent brings us, since it's an empty interface.